test/parallel/test-crypto-webcrypto-aes-decrypt-tag-too-small.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto').webcrypto;
+
+crypto.subtle.importKey(
+  'raw',
+  new Uint8Array(32),
+  {
+    name: 'AES-GCM'
+  },
+  false,
+  [ 'encrypt', 'decrypt' ])
+  .then((k) => {
+    assert.rejects(async () => {
```

Comment:
This function does not need the `async` keyword, and using it hides whether the exception is `throw`n synchronously or the `Promise` is actually rejected.

Reply:
Hmmmmm, Chrome rejects this situation in `Promise`. So shall we do `reject` or `throw`?

Reply:
```js
window.crypto.subtle.decrypt({name: 'AES-GCM', iv: new Uint8Array(12)}, k, new Uint8Array(0));
> Promise {<pending>}
Uncaught (in promise) DOMException: The provided data is too small
```

Reply:
I believe that the behavior is correct, and the test works, but it isn't as strict as it could be. `assert.rejects` is fine, but the `async` weakens the test:

```js
function fnThatThrows() { throw new Error(); }
async function fnThatRejects() { throw new Error(); }

// With 'async':

assert.rejects(async () => fnThatRejects()); // passes
assert.rejects(async () => fnThatThrows()); // passes, but should not!

// Now remove the 'async' keyword:

assert.rejects(() => fnThatRejects()); // passes
assert.rejects(() => fnThatThrows()); // fails as it should
```

So I'd simply remove the `async` keyword from the function declaration :)

Reply:
I've resolved it.